### PR TITLE
CAMEL-15260 - activate span in scope manager on exchange begin to all…

### PR DESCRIPTION
**Pull request by mcrmfc**
…ow spans created outside camel context to be included in trace

[ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful subject line and body.
[ ] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
[ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
[ ] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by mcrmfc** [comment]
Not necessarily ready to merge but opening PR to discuss best approach to testing.

**Comment by jam01** [comment]
Ah nice! I was planning to give this a shot as well :) It's good to know there are others interested in this

The problem I figured with simply activating the Spans is that the default ScopeManager is a ThreadLocalScopeManager which will infinitely keep growing references to the Scoped spans. The Scopes need to be consequently closed. I don't know if the thread local manager is the best fit for camel though.

One alternative is to create processor interceptors that will activate and close scopes before and after every processor, this will allow us to do `GlobalTracer.get().activeSpan()` which would also allow other OpenTracing instrumentation to work and link to the camel created Spans. This is how instrumentation for Mule 4 works.

File: components/camel-opentracing/src/test/java/org/apache/camel/opentracing/CustomSpanAdder.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.apache.camel.opentracing;
```
Comment (oscerd) [inline components/camel-opentracing/src/test/java/org/apache/camel/opentracing/CustomSpanAdder.java:1]: license header is missing

Comment (mcrmfc) [reply components/camel-opentracing/src/test/java/org/apache/camel/opentracing/CustomSpanAdder.java:17]: good spot, added

**Comment by mcrmfc** [comment]
> Ah nice! I was planning to give this a shot as well :) It's good to know there are others interested in this
> 
> The problem I figured with simply activating the Spans is that the default ScopeManager is a ThreadLocalScopeManager which will infinitely keep growing references to the Scoped spans. The Scopes need to be consequently closed. I don't know if the thread local manager is the best fit for camel though.
> 
> One alternative is to create processor interceptors that will activate and close scopes before and after every processor, this will allow us to do `GlobalTracer.get().activeSpan()` which would also allow other OpenTracing instrumentation to work and link to the camel created Spans. This is how instrumentation for Mule 4 works.



> Ah nice! I was planning to give this a shot as well :) It's good to know there are others interested in this
> 
> The problem I figured with simply activating the Spans is that the default ScopeManager is a ThreadLocalScopeManager which will infinitely keep growing references to the Scoped spans. The Scopes need to be consequently closed. I don't know if the thread local manager is the best fit for camel though.
> 
> One alternative is to create processor interceptors that will activate and close scopes before and after every processor, this will allow us to do `GlobalTracer.get().activeSpan()` which would also allow other OpenTracing instrumentation to work and link to the camel created Spans. This is how instrumentation for Mule 4 works.

@jam01 - what are your thoughts as to passing the `scope` returned from `activate` to the `ActiveSpanManager.activate()` method, then we could store in the existing `Holder` and close in the `ActiveSpanManager.deactivate()` method this way we are just ensuring the global tracer is in sync with the internal span activation state?

**Comment by mcrmfc** [comment]
Closing in favour of @jam01 #4001 PR

**Pull request closed without merging**